<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>trackViewer_workshop2020 • trackViewerBiocAsia2020Workshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="trackViewer_workshop2020">
<meta property="og:description" content="trackViewerBiocAsia2020Workshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">trackViewerBiocAsia2020Workshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/trackViewer.html">trackViewer_workshop2020</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jianhong/trackViewerBiocAsia2020Workshop">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="trackViewer_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>trackViewer_workshop2020</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jianhong/trackViewerBiocAsia2020Workshop/blob/master/vignettes/trackViewer.Rmd"><code>vignettes/trackViewer.Rmd</code></a></small>
      <div class="hidden name"><code>trackViewer.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The lollipop-style or needle-style plot are often used for displaying the distribution of mutations or genetic variations along with a variety of genomic annotations, such as gene models and CpG island. Several tools provide lollipop-style plot to show the mutation data such as cBioPortal Tools::MutationMapper<span class="citation"><sup>1</sup></span>, EMBL-EBI::Pfam<span class="citation"><sup>2</sup></span>, BioJS::muts-needle-plot<span class="citation"><sup>3</sup></span>, BiQ Analyzer<span class="citation"><sup>4</sup></span>, and Methylation plotter<span class="citation"><sup>5</sup></span>. The cBioPortal Tools::MutationMapper is a well-known and easy to use on-line genome browser that can output high quality of figures with mutations with tab-delimited mutation data. In R/Bioconductor, there are options to use the flexibility of the Rgraphics system to display Methylation data such as <em>MethVisual</em><span class="citation"><sup>6</sup></span> and <em>GenVisR</em><span class="citation"><sup>7</sup></span>.</p>
<p><strong>Table 1: Tools availble for lollipop plot. </strong></p>
<table class="table">
<colgroup>
<col width="44%">
<col width="19%">
<col width="7%">
<col width="28%">
</colgroup>
<thead><tr class="header">
<th align="left">software</th>
<th align="left">inputs</th>
<th align="left">online</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">MutationMapper<span class="citation"><sup>1</sup></span>
</td>
<td align="left">tab-delimited text</td>
<td align="left">Yes</td>
<td align="left">interprets mutations with different heights along protein annotations in automatic color theme</td>
</tr>
<tr class="even">
<td align="left">Pfam<span class="citation"><sup>2</sup></span>
</td>
<td align="left">JSON</td>
<td align="left">Yes</td>
<td align="left">could combine different line and head colors with different drawing styles</td>
</tr>
<tr class="odd">
<td align="left">muts-needle-plot<span class="citation"><sup>3</sup></span>
</td>
<td align="left">JSON</td>
<td align="left">No</td>
<td align="left">plots data point with different colors, heights, and size along annotations, and highlighted selcted coordinates</td>
</tr>
<tr class="even">
<td align="left">BiQ Analyzer<span class="citation"><sup>4</sup></span>
</td>
<td align="left">BiQ methylation file</td>
<td align="left">Yes</td>
<td align="left">interprets methylation status in black &amp; white</td>
</tr>
<tr class="odd">
<td align="left">Methylation plotter<span class="citation"><sup>5</sup></span>
</td>
<td align="left">tab-delimited text</td>
<td align="left">Yes</td>
<td align="left">stacked multiple methylation status for multiple samples</td>
</tr>
<tr class="even">
<td align="left">
<em>MethVisual</em><span class="citation"><sup>6</sup></span>
</td>
<td align="left">R list</td>
<td align="left">No</td>
<td align="left">visualize the methylation status of CpGs according to their genomic position</td>
</tr>
<tr class="odd">
<td align="left">
<em>REDseq</em><span class="citation"><sup>8</sup></span>
</td>
<td align="left">R list</td>
<td align="left">No</td>
<td align="left">plot frequencies of methylations and SNPs along a chromosome</td>
</tr>
<tr class="even">
<td align="left">
<em>GenVisR</em><span class="citation"><sup>7</sup></span>
</td>
<td align="left">R dataframe</td>
<td align="left">No</td>
<td align="left">plot most accurate graphic representation of the ensembl annotation version based on biomart service</td>
</tr>
</tbody>
</table>
<p>However, when mutations/SNPs are densely distributed, figures generated by the above tools can become cluttered and unreadable.To fill this gap, we developed <em>trackViewer</em>, an R/Bioconductor package as an enhanced light-weight genome viewer for visualizing various types of high-throughput sequencing data, side by side or stacked with multiple annotation tracks. Besides the regular read coverage tracks supported by existing genome browsers/viewers, <em>trackViewer</em> can also be used to generate lollipop and dandeline plot to depict the sparse or dense methylation and SNP/mutation status, together with coverage data and annotation tracks to facilitate integrated analysis of multi-omics data. In addition, figures generated by <em>trackViewer</em> are interactive, i.e., the feel-and-look such as the layout, the color scheme, and the order of tracks can be easily customized by the users. Furthermore, <em>trackViewer</em> can be easily integrated into standard analysis pipeline for various high-throughput sequencing dataset such as ChIP-seq, RNA-seq, methylation-seq or DNA-seq. The images produced by <em>trackViewer</em> are highly customizable including labels, symbols, colors, and size. Here, we illustrate its utilities and capabilities in deriving biological insights from multi-omics dataset from GEO/Encode.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em>BiocManager</em> is used to install the released version of <em>trackViewer</em>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BiocManager</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"trackViewer"</span><span class="op">)</span></code></pre></div>
<p>To check the current version of <em>trackViewer</em>, please try</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">"trackViewer"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] '1.27.14'</code></pre>
</div>
<div id="quick-guide-for-generating-lollipop-plots-using-the-lolliplot-function-in-trackviewer-package-" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-guide-for-generating-lollipop-plots-using-the-lolliplot-function-in-trackviewer-package-" class="anchor"></a>Quick guide for generating lollipop plots using the <em>lolliplot</em> function in <em>trackViewer</em> package.</h2>
<div id="step-1--prepare-the-methylationvariantmutation-data" class="section level3">
<h3 class="hasAnchor">
<a href="#step-1--prepare-the-methylationvariantmutation-data" class="anchor"></a>Step 1. Prepare the methylation/variant/mutation data</h3>
<p>The input data for the <em>lolliplot</em> function in <em>trackViewer</em> is an object of <em>GenomicRanges</em>::<code>GRanges</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">trackViewer</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co">## set the seed to make sure the analysis can be repeated.</span>
<span class="co">## SNP locations to be randomly sampled</span>
<span class="va">SNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">100</span>, <span class="fl">105</span>, <span class="fl">108</span>, <span class="fl">400</span>, <span class="fl">410</span>, <span class="fl">420</span>, <span class="fl">600</span>, <span class="fl">700</span>, <span class="fl">805</span>, <span class="fl">840</span>, <span class="fl">1400</span>, <span class="fl">1402</span><span class="op">)</span>
<span class="co">## use GenomicRanges::GRanges function to create a GRanges object.</span>
<span class="co">## For real data, users can import vcf data via VariantAnnotation::readVcf function.</span>
<span class="va">sample.gr</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="va">SNP</span>, width<span class="op">=</span><span class="fl">1</span>, 
                     <span class="co">## the name of GRanges object will be used as label</span>
                     names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"snp"</span>, <span class="va">SNP</span><span class="op">)</span><span class="op">)</span>,
                     <span class="co">## the score value will be used as the height of lollipop</span>
                     score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample.int</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">SNP</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                     <span class="co">## set the color for lollipop node.</span>
                     color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample.int</a></span><span class="op">(</span><span class="fl">6</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">SNP</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                     <span class="co">## set the color fot the lollipop stem</span>
                     border <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"gray80"</span>, <span class="st">"gray30"</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">SNP</span><span class="op">)</span>, replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
                     <span class="op">)</span></code></pre></div>
</div>
<div id="step-2--prepare-the-geneprotein-model" class="section level3">
<h3 class="hasAnchor">
<a href="#step-2--prepare-the-geneprotein-model" class="anchor"></a>Step 2. Prepare the gene/protein model</h3>
<p>The gene/protein model for the <em>lolliplot</em> function by <em>trackViewer</em> is also an object of <em>GenomicRanges</em>::<code>GRanges</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">features.gr</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">501</span>, <span class="fl">1001</span><span class="op">)</span>, 
                                       width<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">120</span>, <span class="fl">400</span>, <span class="fl">405</span><span class="op">)</span>,
                                       names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"exon"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span>,
                       fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#FF8833"</span>, <span class="st">"#51C6E6"</span>, <span class="st">"#DFA32D"</span><span class="op">)</span>, <span class="co">## color for exon</span>
                       height <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.02</span>, <span class="fl">0.05</span>, <span class="fl">0.08</span><span class="op">)</span> <span class="co">## height for exon</span>
                    <span class="op">)</span></code></pre></div>
</div>
<div id="step-3--plot-lollipop-plot-using-the-lolliplot-function" class="section level3">
<h3 class="hasAnchor">
<a href="#step-3--plot-lollipop-plot-using-the-lolliplot-function" class="anchor"></a>Step 3. Plot lollipop plot using the <em>lolliplot</em> function</h3>
<p>Densely distributed SNPs/mutations is displayed side-by-side with jittered positions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/lolliplot.html">lolliplot</a></span><span class="op">(</span><span class="va">sample.gr</span>, <span class="va">features.gr</span><span class="op">)</span></code></pre></div>
<p><img src="trackViewer_files/figure-html/quickStartStep3-1.png" width="576"></p>
</div>
</div>
<div id="plot-data-in-variant-call-format-vcf" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-data-in-variant-call-format-vcf" class="anchor"></a>Plot data in Variant Call Format (VCF)</h2>
<p>VCF is a text file format is a text format for storing gene sequence variations such as genomic positions, original genotypes, and optional genotypes. The <em>trackViewer</em> package can be used to display single nucleotide polymorphisms (SNPs) in VCF file as lollipop-style plots. The input data for the folloiwng example are variant data from chromosome 22 in the 1000 Genomes project which is included in the <em>VariantAnnotation</em> package.</p>
<p>Note that the number of circles indicates the number of SNP events. Different colors differentiate known SNPs from new mutation events.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span> <span class="co">## load the package for reading the vcf file</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span> <span class="co">## load the package for obtaining the gene model</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span> <span class="co">## load the package for fetching the gene name</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span>
<span class="va">fl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"chr22.vcf.gz"</span>, package<span class="op">=</span><span class="st">"VariantAnnotation"</span><span class="op">)</span>
<span class="co">## set the track range</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"22"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="fl">50968014</span>, <span class="fl">50970514</span>, names<span class="op">=</span><span class="st">"TYMP"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## read in the vcf file</span>
<span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu">TabixFile</span><span class="op">(</span><span class="va">fl</span><span class="op">)</span>
<span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html">readVcf</a></span><span class="op">(</span><span class="va">fl</span>, <span class="st">"hg19"</span>, param<span class="op">=</span><span class="va">gr</span><span class="op">)</span>
<span class="co">## get GRanges from the VCF object </span>
<span class="va">mutation.frequency</span> <span class="op">&lt;-</span> <span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>
<span class="co">## keep the metadata</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">mutation.frequency</span><span class="op">)</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">mutation.frequency</span><span class="op">)</span>, 
          <span class="fu">VariantAnnotation</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">info</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">)</span>
<span class="co">## set colors</span>
<span class="va">mutation.frequency</span><span class="op">$</span><span class="va">border</span> <span class="op">&lt;-</span> <span class="st">"gray30"</span>
<span class="va">mutation.frequency</span><span class="op">$</span><span class="va">color</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"^rs"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">mutation.frequency</span><span class="op">)</span><span class="op">)</span>, 
           <span class="st">"lightcyan"</span>, <span class="st">"lavender"</span><span class="op">)</span>
<span class="co">## plot Global Allele Frequency based on AC/AN</span>
<span class="va">mutation.frequency</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">mutation.frequency</span><span class="op">$</span><span class="va">AF</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>
<span class="co">## change the rotation angle of the SNP label</span>
<span class="va">mutation.frequency</span><span class="op">$</span><span class="va">label.parameter.rot</span> <span class="op">&lt;-</span> <span class="fl">45</span>
<span class="co">## keep the sequence level style to UCSC</span>
<span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">mutation.frequency</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span>
<span class="fu">seqlevels</span><span class="op">(</span><span class="va">mutation.frequency</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chr22"</span>
<span class="co">## extract transcripts in the given ranges</span>
<span class="va">trs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/geneModelFromTxdb.html">geneModelFromTxdb</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>, 
                         <span class="va">org.Hs.eg.db</span>, gr<span class="op">=</span><span class="va">gr</span><span class="op">)</span>
<span class="co">## subset the features to show the transcripts of your interest only</span>
<span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">trs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">trs</span><span class="op">[[</span><span class="fl">5</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span><span class="op">)</span>
<span class="co">## define the feature labels</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">trs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">name</span>, <span class="va">trs</span><span class="op">[[</span><span class="fl">5</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>
<span class="co">## define the feature colors</span>
<span class="va">features</span><span class="op">$</span><span class="va">fill</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lightblue"</span>, <span class="st">"mistyrose"</span><span class="op">)</span>
<span class="co">## define the feature heights</span>
<span class="va">features</span><span class="op">$</span><span class="va">height</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.02</span>, <span class="fl">.04</span><span class="op">)</span>
<span class="co">## set the legends</span>
<span class="va">legends</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"known"</span>, <span class="st">"unkown"</span><span class="op">)</span>, 
                fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lightcyan"</span>, <span class="st">"lavender"</span><span class="op">)</span>, 
                color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gray80"</span>, <span class="st">"gray80"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## lollipop plot</span>
<span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/lolliplot.html">lolliplot</a></span><span class="op">(</span><span class="va">mutation.frequency</span>, 
          <span class="va">features</span>, ranges<span class="op">=</span><span class="va">gr</span>, type<span class="op">=</span><span class="st">"circle"</span>, 
          legend<span class="op">=</span><span class="va">legends</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="trackViewer_files/figure-html/plotVCFdata-1.png" alt="lollipop plot for VCF data" width="864"><p class="caption">
lollipop plot for VCF data
</p>
</div>
</div>
<div id="plot-methylation-data-from-files-in-bed-format" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-methylation-data-from-files-in-bed-format" class="anchor"></a>Plot methylation data from files in bed format</h2>
<p>Any data that can be imported into a <code>GRanges</code> object can be viewed by <em>trackViewer</em> package. In the following example, we will illustrate how to plot methylation data in bed format.</p>
<p>First, the <em>rtracklayer</em> package is used to import the methylation data into a <code>GRanges</code>. Then, the transcripts are extracted from the <code>TxDb</code> object with gene symbol annotated using the <code>org</code> database. In addition, multiple transcripts can be shown in different colors and tracks.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span> <span class="co">## load the package for obtaining the gene model</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span> <span class="co">## load the package for fetching the gene name</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span>
<span class="co">## set the track range</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chr22"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="fl">50968014</span>, <span class="fl">50970514</span>, names<span class="op">=</span><span class="st">"TYMP"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## extract transcripts in given ranges</span>
<span class="va">trs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/geneModelFromTxdb.html">geneModelFromTxdb</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>, 
                         <span class="va">org.Hs.eg.db</span>, gr<span class="op">=</span><span class="va">gr</span><span class="op">)</span>
<span class="co">## subset the features to show the transcripts of your interest</span>
<span class="va">features</span> <span class="op">&lt;-</span> <span class="fu">GRangesList</span><span class="op">(</span><span class="va">trs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">dat</span>, <span class="va">trs</span><span class="op">[[</span><span class="fl">5</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">dat</span>, <span class="va">trs</span><span class="op">[[</span><span class="fl">6</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span>
<span class="va">flen</span> <span class="op">&lt;-</span> <span class="fu">elementNROWS</span><span class="op">(</span><span class="va">features</span><span class="op">)</span>
<span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span>
<span class="co">## define feature track layers</span>
<span class="va">features</span><span class="op">$</span><span class="va">featureLayerID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">flen</span><span class="op">[</span><span class="op">-</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>, <span class="va">flen</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">## define feature labels</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">features</span><span class="op">$</span><span class="va">symbol</span>
<span class="co">## define feature colors</span>
<span class="va">features</span><span class="op">$</span><span class="va">fill</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"lightblue"</span>, <span class="st">"mistyrose"</span>, <span class="st">"mistyrose"</span><span class="op">)</span>, <span class="va">flen</span><span class="op">)</span>
<span class="co">## define feature heights</span>
<span class="va">features</span><span class="op">$</span><span class="va">height</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">features</span><span class="op">$</span><span class="va">feature</span><span class="op">==</span><span class="st">"CDS"</span>, <span class="fl">.04</span>, <span class="fl">.02</span><span class="op">)</span>
<span class="co">## import methylation data from a file in bed format</span>
<span class="va">methy</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"methy.bed"</span>, package<span class="op">=</span><span class="st">"trackViewer"</span><span class="op">)</span>, <span class="st">"BED"</span><span class="op">)</span>
<span class="co">## filter the data</span>
<span class="va">methy</span> <span class="op">&lt;-</span> <span class="va">methy</span><span class="op">[</span><span class="va">methy</span><span class="op">$</span><span class="va">score</span> <span class="op">&gt;</span> <span class="fl">20</span><span class="op">]</span>
<span class="co">## Please note that for pie plot, there must be at least two numeric columns</span>
<span class="va">methy</span><span class="op">$</span><span class="va">score2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">methy</span><span class="op">$</span><span class="va">score</span><span class="op">)</span> <span class="op">-</span> <span class="va">methy</span><span class="op">$</span><span class="va">score</span>
<span class="co">## set the legends</span>
<span class="va">legends</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"methylated"</span>, <span class="st">"unmethylated"</span><span class="op">)</span>, 
                fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"lightblue"</span><span class="op">)</span>, 
                color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"black"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## lollipop plot in pie layout</span>
<span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/lolliplot.html">lolliplot</a></span><span class="op">(</span><span class="va">methy</span>, <span class="va">features</span>, 
          ranges<span class="op">=</span><span class="va">gr</span>, type<span class="op">=</span><span class="st">"pie"</span>, 
          legend<span class="op">=</span><span class="va">legends</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="trackViewer_files/figure-html/plotMethyBed-1.png" alt="lollipop plot, pie layout" width="864"><p class="caption">
lollipop plot, pie layout
</p>
</div>
</div>
<div id="plot-lollipop-plot-for-multiple-patients-in-pie-stack-layout" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-lollipop-plot-for-multiple-patients-in-pie-stack-layout" class="anchor"></a>Plot lollipop plot for multiple patients in “pie.stack” layout</h2>
<p>The percentage of methylation rates can be shown in pie chart with different layers for different patients.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## simulate multiple patients</span>
<span class="va">rand.id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample.int</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">methy</span><span class="op">)</span>, <span class="fl">3</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">methy</span><span class="op">)</span>, replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">rand.id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">rand.id</span><span class="op">)</span>
<span class="va">methy.mul.patient</span> <span class="op">&lt;-</span> <span class="va">methy</span><span class="op">[</span><span class="va">rand.id</span><span class="op">]</span>
<span class="co">## Please note that pie.stack requires metadata "stack.factor", score, and score2. In addtion, </span>
<span class="co">## the metadata cannot contain stack.factor.order or stack.factor.first for pie.stack style.</span>
<span class="va">len.max</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">rand.id</span><span class="op">)</span><span class="op">)</span>
<span class="va">stack.factors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"patient"</span>, 
                        <span class="fu"><a href="https://rdrr.io/r/base/formatc.html">formatC</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">len.max</span>, 
                                width<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">len.max</span><span class="op">)</span><span class="op">)</span>, 
                                flag<span class="op">=</span><span class="st">"0"</span><span class="op">)</span><span class="op">)</span>
<span class="va">methy.mul.patient</span><span class="op">$</span><span class="va">stack.factor</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">rand.id</span><span class="op">)</span>, <span class="va">sample</span>, x<span class="op">=</span><span class="va">stack.factors</span><span class="op">)</span><span class="op">)</span>
<span class="va">methy.mul.patient</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample.int</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">methy.mul.patient</span><span class="op">)</span>, replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## for a pie plot, two or more numeric meta-columns are required.</span>
<span class="va">methy.mul.patient</span><span class="op">$</span><span class="va">score2</span> <span class="op">&lt;-</span> <span class="fl">100</span> <span class="op">-</span> <span class="va">methy.mul.patient</span><span class="op">$</span><span class="va">score</span>
<span class="co">## set different color set for different patient</span>
<span class="va">patient.color.set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html">rainbow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">stack.factors</span><span class="op">)</span><span class="op">)</span>, 
                                                 <span class="st">"#FFFFFFFF"</span><span class="op">)</span>, 
                                           stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">patient.color.set</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">stack.factors</span>
<span class="va">methy.mul.patient</span><span class="op">$</span><span class="va">color</span> <span class="op">&lt;-</span> 
    <span class="va">patient.color.set</span><span class="op">[</span><span class="va">methy.mul.patient</span><span class="op">$</span><span class="va">stack.factor</span><span class="op">]</span>
<span class="co">## set the legends</span>
<span class="va">legends</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>labels<span class="op">=</span><span class="va">stack.factors</span>, col<span class="op">=</span><span class="st">"gray80"</span>, 
                fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">patient.color.set</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="co">## lollipop plot</span>
<span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/lolliplot.html">lolliplot</a></span><span class="op">(</span><span class="va">methy.mul.patient</span>, 
          <span class="va">features</span>, ranges<span class="op">=</span><span class="va">gr</span>, 
          type<span class="op">=</span><span class="st">"pie.stack"</span>, 
          legend<span class="op">=</span><span class="va">legends</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="trackViewer_files/figure-html/plotMethypieStack-1.png" alt="lollipop plot, pie.stack layout" width="864"><p class="caption">
lollipop plot, pie.stack layout
</p>
</div>
</div>
<div id="plot-lollipop-plot-in-caterpillar-layout-to-faciliate-the-comparison-between-two-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-lollipop-plot-in-caterpillar-layout-to-faciliate-the-comparison-between-two-samples" class="anchor"></a>Plot lollipop plot in caterpillar layout to faciliate the comparison between two samples</h2>
<p>The caterpillar layout can be used to visualize multiple samples side by side or to display dense data.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## use SNPsideID to separate samples or densely distributed SNPs/mutations/methylation events</span>
<span class="va">sample.gr</span><span class="op">$</span><span class="va">SNPsideID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"top"</span>, <span class="st">"bottom"</span><span class="op">)</span>, 
                               <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">sample.gr</span><span class="op">)</span>,
                               replace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/lolliplot.html">lolliplot</a></span><span class="op">(</span><span class="va">sample.gr</span>, <span class="va">features.gr</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="trackViewer_files/figure-html/plotCaterpillar-1.png" alt="lollipop plot, caterpillar layout" width="864"><p class="caption">
lollipop plot, caterpillar layout
</p>
</div>
</div>
<div id="dandelion-plot-for-the-visualization-of-data-with-densely-distributed-snpsmutationsmethylation-events-" class="section level2">
<h2 class="hasAnchor">
<a href="#dandelion-plot-for-the-visualization-of-data-with-densely-distributed-snpsmutationsmethylation-events-" class="anchor"></a>Dandelion plot for the visualization of data with densely distributed SNPs/mutations/methylation events.</h2>
<p>Sometimes, there are as many as hundreds of SNPs or methylation status in one gene. Dandelion plot can be used to depict such dense SNPs or methylation events. Please note that the height of the dandelion indicates the density of the events.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">methy</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"methy.bed"</span>, package<span class="op">=</span><span class="st">"trackViewer"</span><span class="op">)</span>, <span class="st">"BED"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">methy</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 38</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## set the color of dandelion leaves.</span>
<span class="va">methy</span><span class="op">$</span><span class="va">color</span> <span class="op">&lt;-</span> <span class="fl">3</span>
<span class="va">methy</span><span class="op">$</span><span class="va">border</span> <span class="op">&lt;-</span> <span class="st">"gray"</span>
<span class="co">## The following normalization assumes that the total number of events (methy + unmethy) are the same across differnt genomic locations and the ones with the highest score has 100% methylation rate. This is for illustation purpose. For your own data, you will need to calculate the score using methylated/(methylated + unmethylated) for each position.</span>
<span class="co">## rescale the methylation score using the max value of the score </span>
<span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">methy</span><span class="op">$</span><span class="va">score</span><span class="op">)</span>
<span class="va">methy</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="va">methy</span><span class="op">$</span><span class="va">score</span><span class="op">/</span><span class="va">m</span>
<span class="co"># The area of the fan indicates the percentage of methylation or rate of mutation.</span>
<span class="fu"><a href="https://rdrr.io/pkg/trackViewer/man/dandelion.plot.html">dandelion.plot</a></span><span class="op">(</span><span class="va">methy</span>, <span class="va">features</span>, ranges<span class="op">=</span><span class="va">gr</span>, type<span class="op">=</span><span class="st">"fan"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="trackViewer_files/figure-html/dandelionPlot-1.png" alt="dandelion plot" width="576"><p class="caption">
dandelion plot
</p>
</div>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-gao2013integrative" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Gao, J. <em>et al.</em> Integrative analysis of complex cancer genomics and clinical profiles using the cBioPortal. <em>Science signaling</em> <strong>6,</strong> pl1 (2013).</div>
</div>
<div id="ref-finn2006pfam" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Finn, R. D. <em>et al.</em> Pfam: Clans, web tools and services. <em>Nucleic acids research</em> <strong>34,</strong> D247–D251 (2006).</div>
</div>
<div id="ref-michael_p_schroeder_2015_14561" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Schroeder, M. P. Muts-needle-plot: Mutations needle plot v0.8.0. (2015). doi:<a href="https://doi.org/10.5281/zenodo.14561">10.5281/zenodo.14561</a>
</div>
</div>
<div id="ref-bock2005biq" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Bock, C. <em>et al.</em> BiQ analyzer: Visualization and quality control for DNA methylation data from bisulfite sequencing. <em>Bioinformatics</em> <strong>21,</strong> 4067–4068 (2005).</div>
</div>
<div id="ref-mallona2014methylation" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">Mallona, I., Dı́ez-Villanueva, A. &amp; Peinado, M. A. Methylation plotter: A web tool for dynamic visualization of DNA methylation data. <em>Source code for biology and medicine</em> <strong>9,</strong> 1 (2014).</div>
</div>
<div id="ref-zackay2010methvisual" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">Zackay, A. &amp; Steinhoff, C. MethVisual-visualization and exploratory statistical analysis of DNA methylation profiles from bisulfite sequencing. <em>BMC research notes</em> <strong>3,</strong> 337 (2010).</div>
</div>
<div id="ref-skidmore2016genvisr" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">Skidmore, Z. L. <em>et al.</em> GenVisR: Genomic visualizations in r. <em>Bioinformatics</em> btw325 (2016).</div>
</div>
<div id="ref-zhu4redseq" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">Zhu, L. &amp; others. REDseq: A bioconductor package for analyzing high throughput sequencing data from restriction enzyme digestion.</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jianhong Ou, Lihua Julie Zhu Developer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
